### ISSUE TITLE
Gulp-jest doesn't apply config (moduleNameMapper to be specific)

### ISSUE THREAD

**Issue by deFarro** [alansouzati/gulp-jest#40]
For some reason moduleNameMapper works for "npm test" but doesn't work for "gulp test".

All the details here:
https://stackoverflow.com/questions/45945522/gulp-jest-doesnt-apply-config-modulenamemapper-to-be-specific

Project is here:
https://github.com/deFarro/gulp-jest-issue

**Comment by shanimal** [comment]
I've noticed if I run *jest* from the command line the configs are picked up automatically...

    // package.json
    "scripts": {
        "jest": "jest",
        "test": "gulp test"
    }

    > npm run jest √
    > npm test √

but the runCLI method needs to have the config passed in...

    // config in package.json
    import { jest as jestConfig } from '../../package.json';
    // config in jest.config.js
    import jestConfig from '../../jest.config';

    jestCLI.runCLI({ config }, [rootDir], (success) => {})

should it also pick it up automatically?

**Comment by deFarro** [comment]
Yep, but've passed config to jest directly in gulpfile this way:

```
return gulp.src('test')
    .pipe(jest({
      config: {
        "transformIgnorePatterns": [
          "<rootDir>/dist/", "<rootDir>/node_modules/"
        ],
        "automock": false,
        "moduleNameMapper": {
          "\\.scss$": "<rootDir>/jestignore.js"
        }
      }
    }));
```

And this way:

```
return gulp.src('test')
    .pipe(jest(require('./jest.config.js')));
```

And both ways it's ignoring my config when I run it with "gulp test", while "jest" command works fine.
So I can't add jest to my webpack workflow, to run all test automatically before building a bundle.

**Comment by anselmbradford** [comment]
I found the jest-cli 20.0.4 loads the config with `.pipe(jest({ config: { … } }))`, but for any higher version it doesn't appear to apply the config.

**Comment by anselmbradford** [comment]
I went with calling jest-cli directly with:

```js
const spawn = require( 'child_process' ).spawn;

 spawn(
    'path/to/node_modules/jest-cli/bin/jest.js',
    ['--config=./test/config-jest.json'],
    { stdio: 'inherit' }
  ).once( 'close', code => {
    if ( code ) {
      console.log( 'Tests exited with code ' + code );
      process.exit( 1 );
    }
    console.log( 'Tests done!' );
  } );
```

And the config (`test/config-jest.json`):

```
{
  "verbose": true,
   … your config options here
}
```

**Comment by alansouzati** [comment]
let me take a look on this. sorry I was super busy recently.

**Comment by alansouzati** [comment]
I believe I fixed this.

The default rootDir if you did not pass one was `file.path`. I've changed it to `process.cwd()`. Now the configuration in your jestignore should be picked up.

I added a test for this also.

Here is the commit https://github.com/alansouzati/gulp-jest/commit/403dcdf1bbd404d772a8187fa2e374c495e341c9

@deFarro Can you try in your repo to see if this has been fixed?

**Comment by deFarro** [comment]
Yep, I checked latest version with my other projects - it works properly.
Thank you!
